client/src/components/App/index.tsx

```diff
@@ -206,13 +191,17 @@ export const AppComponent: FunctionComponent<AppProps> = observer(props => {
       isStreaming={isStreaming}
       namespaces={store.controls.namespaces}
       currentNamespace={store.controls.currentNamespace}
-      onNamespaceChange={onNamespaceChange}
+      onNamespaceChange={store.controls.setCurrentNamespace}
       selectedVerdict={store.controls.verdict}
-      onSelectVerdict={onSelectVerdict}
+      onVerdictChange={store.controls.setVerdict}
       selectedHttpStatus={store.controls.httpStatus}
-      onSelectHttpStatus={onSelectHttpStatus}
+      onHttpStatusChange={store.controls.setHttpStatus}
       flowFilters={store.controls.flowFilters}
-      onChangeFlowFilters={onChangeFlowFilters}
+      onChangeFlowFilters={store.setFlowFilters}
+      showHost={store.controls.showHost}
+      onShowHostToggle={store.controls.toggleShowHost}
```

Comment:
This is beautiful line, but it seems like logic should be extended to explicitly drop the data on toggle:
```ts
const toggleShowHost = useCallback(() => {
  const isActive = store.controls.toggleShowHost();
  if (!isActive) return;

  store.clearMap();
}, []);
```

This comment is also related to your [reaction](https://github.com/cilium/hubble-ui/pull/72/files#diff-ba7be394959cb356cebbdafbe33679eaR230) in main store, which is not about big explicitness